src/rules/noInvalidTemplateStringsRule.ts

```diff
@@ -50,24 +50,18 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 
     function check(node: ts.StringLiteral): void {
-        /**
-         * Finds instances of '${'
-         */
-        const findTemplateString = new RegExp(/\$\{/);
-            
-        const index = node.text.search(findTemplateString);
-        if (index !== -1) {
-            /**
-             * Support for ignoring case: '\${template-expression}'
-             */
-            const unescapedText = node.getFullText();
-            const preceedingCharacter = unescapedText.substr(unescapedText.search(findTemplateString) - 1, 1);
-            if (isBackslash(preceedingCharacter)) {
-                return;
+        const text = node.getText(ctx.sourceFile);
+        const findTemplateStrings = /\$\{/g;
+        const nextInstance = () => instance = findTemplateStrings.exec(text);
+        let instance: RegExpExecArray | null = nextInstance();
+        while (instance !== null) {
+            const preceedingCharacters = text.substr(instance.index - 2, 2);
```

Comment:
better count the preceding backslashes. If it's an odd number, the `$` is escaped with a `\`

consider: `"\\\${}"` is escaped, but `"\\\\${}"` is not